tests/__init__.py

```diff
@@ -34,13 +34,23 @@
 from s3transfer.futures import TransferMeta
 from s3transfer.futures import TransferFuture
 from s3transfer.futures import BoundedExecutor
+from s3transfer.futures import NonThreadedExecutor
 from s3transfer.subscribers import BaseSubscriber
 from s3transfer.utils import OSUtils
 from s3transfer.utils import CallArgs
 from s3transfer.utils import TaskSemaphore
 from s3transfer.utils import SlidingWindowSemaphore
 
 
+def setup_package():
```

Comment:
While not strictly necessary, I still think it's good practice to put back any global state you mutated.  I think we should have a `teardown_package()` as well.